clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -118,49 +123,78 @@ public PrototypeAsyncConsumer(final Map<String, Object> configs,
     public PrototypeAsyncConsumer(final ConsumerConfig config,
                                   final Deserializer<K> keyDeserializer,
                                   final Deserializer<V> valueDeserializer) {
-        this.time = Time.SYSTEM;
+        this(Time.SYSTEM, config, keyDeserializer, valueDeserializer);
+    }
+
+
+    public PrototypeAsyncConsumer(final Time time,
```

Comment:
Not introduced by this PR, but concern regarding this constructor not handling creation errors and potential resource leak. That is specifically handled in the current consumer implementation after some issues (see [here](https://github.com/apache/kafka/blob/b10beaae77905e6750a789ed52ecddf8ff81a9a6/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L831)). I don't have the full understanding of the underlying issue but worth reviewing/discussing for this case 
(not looking for any specific change on this PR. Just checking if there's been already some discussion about it for the PrototypeAsyncConsumer, if not I'll track it for discussion)

Reply:
I'm not aware of any such discussions. Can you file a Jira or bring it up somewhere? Thanks!